src/java.base/share/classes/com/sun/crypto/provider/AESParameters.java

```diff
@@ -46,7 +47,7 @@
     private BlockCipherParamsCore core;
 
     public AESParameters() {
-        core = new BlockCipherParamsCore(AESConstants.AES_BLOCK_SIZE);
+        core = new BlockCipherParamsCore(AESConstants.AES_BLOCK_SIZE, 4, 8);
```

Comment:
A cipher object may not take different IV sizes at the same time.  I was just wondering how it could be used in practice.  Maybe something like:

```
AlgorithmParameters algParams = AlgorithmParameters.getInstance("AES");
algParams.init(ivParameterSpec);
```
The IV parameter is given with the init() method.  Then, it may be not necessary to construct the BlockCipherParamsCore object will all potential IV sized.  See the comments in BlockCipherParamsCore.

Reply:
Cipher objects normally takes just one iv size. BlockCipherParamsCore is used by various impls of AlgorithmParametersSpi class which may be used with different block cipher algorithms. The iv parameter is given with the AlgorithmParametersSpi.init() method and invalid iv will lead to exceptions. Since there are iv size checks built in BlockCipherParamsCore already, it seems safer to relax the check a bit for AES (4 and 8 for KWP and KW). The other choice is to just remove the size check from BlockCipherParamsCore for AES algorithm, but then we'd not be able to reject invalid iv values as before.